src/rules.js

```diff
@@ -1699,6 +2163,289 @@ module.exports = [
             "transform": "$0"
         }
     },
+    // transform-origin
+    {
+        "type": "pattern",
+        "name": "Transform origin",
+        "matcher": "Trfo",
+        "allowParamToValue": true,
+        "styles": {
+            "transform-origin": "$0 $1"
+        },
+        "arguments": [{
+            "t": "top",
+            "end": "__END__",
+            "bottom": "bottom",
+            "start": "__START__",
+            "c": "center"
+        },{
+            "t": "top",
+            "end": "__END__",
+            "bottom": "bottom",
+            "start": "__START__",
+            "c": "center"
+        }]
+    },
+    // transform-style
+    {
+        "type": "pattern",
+        "name": "Transform style",
+        "matcher": "Trfs",
+        "allowParamToValue": false,
+        "styles": {
+            "transform-style": "$0"
+        },
+        "arguments": [{
+            "f": "flat",
+            "p": "preserve-3d"
+        }]
+    },
+    // perspective
+    {
+        "type": "pattern",
+        "name": "Perspective",
+        "matcher": "Prs",
+        "allowParamToValue": true,
+        "styles": {
+            "perspective": "$0"
+        },
+        "arguments": [{
+            "n": "none"
+        }]
+    },
+    // perspective-origin
+    {
+        "type": "pattern",
+        "name": "Perspective origin",
+        "matcher": "Prso",
+        "allowParamToValue": true,
+        "styles": {
+            "perspective-origin": "$0 $1"
+        },
+        "arguments": [{
+            "t": "top",
+            "end": "__END__",
+            "bottom": "bottom",
+            "start": "__START__",
+            "c": "center"
+        },{
+            "t": "top",
+            "end": "__END__",
+            "bottom": "bottom",
+            "start": "__START__",
+            "c": "center"
+        }]
+    },
+    // backface-visibility
+    {
+        "type": "pattern",
+        "name": "Backface visibility",
+        "matcher": "Bfv",
+        "allowParamToValue": false,
+        "styles": {
+            "backface-visibility": "$0"
+        },
+        "arguments": [{
+            "h": "hidden",
+            "v": "visible"
+        }]
+    },
+    // matrix
+    {
+        "type": "pattern",
+        "name": "Matrix (transform)",
+        "matcher": "Matrix",
+        "allowParamToValue": true,
+        "styles": {
+            "transform": "matrix($0,$1,$2,$3,$4,$5)"
+        }
+    },
+    // matrix3d
+    {
+        "type": "pattern",
+        "name": "Matrix 3d (transform)",
+        "matcher": "Matrix3d",
+        "allowParamToValue": true,
+        "styles": {
+            "transform": "matrix($0,$1,$2,$3,$4,$5,$6,$7,$8,$9,$10,$11,$12,$13,$14,$15)"
```

Comment:
This one will probably look funny in the reference docs... you'll get 15 "value" strings in a row :-/